the-incredible-bulk/src/test/java/gov/va/api/health/bulkfhir/service/controller/patient/BulkPatientControllerTest.java

```diff
@@ -0,0 +1,240 @@
+package gov.va.api.health.bulkfhir.service.controller.patient;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import gov.va.api.health.bulkfhir.service.controller.publication.DefaultPublicationStatusTransformer;
+import gov.va.api.health.bulkfhir.service.status.StatusEntity;
+import gov.va.api.health.bulkfhir.service.status.StatusRepository;
+import gov.va.api.health.dstu2.api.resources.OperationOutcome;
+import gov.va.api.health.ids.api.IdentityService;
+import gov.va.api.health.ids.api.Registration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.ResponseEntity;
+
+@ExtendWith(MockitoExtension.class)
+public class BulkPatientControllerTest {
+
+  @Mock StatusRepository repo;
+
+  @Mock HttpServletRequest request;
+
+  @Mock IdentityService identityService;
+
+  BulkPatientController controller() {
+    return BulkPatientController.builder()
+        .repository(repo)
+        .baseUrl("http://fake-va.gov")
+        .bulkStatusPath("/STATUSNOW")
+        .identityService(identityService)
+        .transformer(new DefaultPublicationStatusTransformer())
+        .build();
+  }
+
+  @Test
+  void
+      exportReturnsContentLocationHeaderForFirstPublicationWhenMultipleCompletedPublicationExist() {
+    when(repo.findDistinctPublicationIds()).thenReturn(List.of("1", "2"));
+    long now = Instant.now().toEpochMilli();
+    when(repo.findByPublicationId("1"))
+        .thenReturn(
+            List.of(
+                StatusEntity.builder()
+                    .publicationId("1")
+                    .publicationEpoch(now)
+                    .buildCompleteEpoch(now)
+                    .buildStartEpoch(now)
+                    .recordsPerFile(10)
+                    .build()));
+    when(request.getRequestURI()).thenReturn("/Patient/$export");
+    when(request.getQueryString()).thenReturn("_outputFormat=ndjson");
+    when(identityService.register(anyList()))
+        .thenReturn(List.of(Registration.builder().uuid("ZZZ").build()));
+    Map<String, String> headers = new HashMap<>();
+    headers.put("accept", "application/fhir+json");
+    headers.put("prefer", "respond-async");
```

Comment:
A helper method might remove some of the duplication of data initialization for these tests.

Reply:
Yep, good call, moved the common header setup to another method.